src/test/java/com/epam/tc/hw2/ex2/DifferentElementsPageTest.java

```diff
@@ -0,0 +1,136 @@
+package com.epam.tc.hw2.ex2;
+
+import com.epam.tc.hw2.BasePageTest;
+import com.epam.tc.hw2.Utils;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.assertj.core.api.SoftAssertions;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.Select;
+import org.testng.annotations.Test;
+
+public class DifferentElementsPageTest extends BasePageTest {
+
+    @Test
+    public void testDifferentElementsPage() {
+        SoftAssertions softAssertions = new SoftAssertions();
+
+        // 1. Open test site by URL
+        openSite();
+
+        // 2. Assert Browser title
+        testBrowserTitle(softAssertions);
+
+        // 3. Perform login
+        performLogin();
+
+        // 4. Assert User name in the left-top side of screen that user is loggined
+        testUsername(softAssertions);
+
+        // 5. Open through the header menu Service -> Different Elements Page
+        openDifferentElementsPage(softAssertions);
+
+        // 6. Select checkboxes Water, Wind
+        selectCheckboxes(softAssertions);
+
+        // 7. Select radio Selen
+        selectRadioButtons(softAssertions);
+
+        // 8. Select in dropdown Yellow
+        selectDropdown(softAssertions);
+
+        // 9. Assert that for each checkbox there is an individual log row
+        // and value is corresponded to the status of checkbox
+        testCheckboxesLog(softAssertions);
+
+        // 9. Assert that for radio button there is a log row
+        // and value is corresponded to the status of radio button
+        testRadioButtonLog(softAssertions);
+
+        // 9. Assert that for dropdown there is a log row
+        // and value is corresponded to the selected value.
+        testDropdownLog(softAssertions);
+
+        softAssertions.assertAll();
+    }
+
+    private void openDifferentElementsPage(SoftAssertions softAssertions) {
+        WebElement service = driver.findElement(By.cssSelector(".nav > li.dropdown"));
+        service.click();
+
+        WebElement link = driverWait.until(ExpectedConditions.elementToBeClickable(By.xpath(
+                String.format(
+                        "//ul[@class=\"dropdown-menu\"]/li/*[contains(text(),\"%s\")]",
+                        Utils.DIFFERENT_ELEMENTS_LINK_NAME))));
```

Comment:
Use just string locator instead of string.format for such cases.

Reply:
Done